src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -206,24 +208,53 @@ public void SuppressOutput()
         }
 
         /// <summary>
-        /// A delegate used to execute children asynchronously.
+        /// Execute children asynchronously and return their rendered content.
         /// </summary>
         /// <returns>A <see cref="Task"/> that on completion returns content rendered by children.</returns>
         /// <remarks>This method is memoized.</remarks>
         public Task<TagHelperContent> GetChildContentAsync()
         {
-            return GetChildContentAsync(useCachedResult: true);
+            return GetChildContentAsync(useCachedResult: true, encoder: null);
         }
 
         /// <summary>
-        /// A delegate used to execute children asynchronously.
+        /// Execute children asynchronously and return their rendered content.
         /// </summary>
         /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
         /// of child content; cached content will be returned.</param>
         /// <returns>A <see cref="Task"/> that on completion returns content rendered by children.</returns>
         public Task<TagHelperContent> GetChildContentAsync(bool useCachedResult)
         {
-            return _getChildContentAsync(useCachedResult);
+            return GetChildContentAsync(useCachedResult, encoder: null);
+        }
+
+        /// <summary>
+        /// Execute children asynchronously with the given <paramref name="encoder"/> in scope and return their
+        /// rendered content.
+        /// </summary>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use. Does not override all HTML encodings which may occur.
+        /// </param>
+        /// <returns>A <see cref="Task"/> that on completion returns content rendered by children.</returns>
+        /// <remarks>This method is memoized.</remarks>
+        public Task<TagHelperContent> GetChildContentAsync(HtmlEncoder encoder)
+        {
+            return GetChildContentAsync(useCachedResult: true, encoder: encoder);
```

Comment:
If encoder is `null` we may need to do something based on [this](https://github.com/aspnet/Razor/pull/653/files#r48931958) conversation. Same with next overload.